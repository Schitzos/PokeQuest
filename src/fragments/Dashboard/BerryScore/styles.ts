import {Platform, StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  expPoint: {
    zIndex: 4000,
    position: 'absolute',
    top: Platform.OS === 'ios' ? -410 : -450,
    backgroundColor: 'rgba(255, 255, 255, 0.8)',
    padding: 8,
    borderRadius: 16,
    left: 250,
  },
});
