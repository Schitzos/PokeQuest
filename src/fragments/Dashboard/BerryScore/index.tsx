import React, {useEffect, useRef} from 'react';
import {Animated} from 'react-native';
import TextView from '@/components/TextView';
import theme from '@/theme';
import {usePokemon} from '@/hooks/usePokemon';
import {useAnimation} from '@/hooks/useAnimation';
import {styles} from './styles';

export default function BerryScore() {
  const {pokemon} = usePokemon();
  const {animateOpacityToggle} = useAnimation();
  const opacityAnimation = new Animated.Value(0);
  const isMounted = useRef(false);

  const oldExp = pokemon?.selected.prevExp!;
  const curExp = pokemon?.selected.currentExp!;
  const expDifference = curExp - oldExp;
  const sign = expDifference >= 0 ? '+' : '';

  const randomY = Math.random() * (400 - 200) - 400;
  const randomX = Math.random() * (250 - 50) + 50;

  useEffect(() => {
    if (isMounted.current) {
      animateOpacityToggle(opacityAnimation, 2000);
    } else {
      isMounted.current = true;
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [pokemon?.selected]);

  return (
    <Animated.View
      style={[
        styles.expPoint,
        {opacity: opacityAnimation, left: randomX, top: randomY},
      ]}>
      <TextView
        color={expDifference >= 0 ? theme.colors.success : theme.colors.error}
        fz={24}>
        {sign}
        {curExp - oldExp}
      </TextView>
    </Animated.View>
  );
}
