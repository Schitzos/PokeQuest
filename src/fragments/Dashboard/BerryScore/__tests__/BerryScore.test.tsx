// BerryScore.test.tsx

import React from 'react';
import {render} from '@testing-library/react-native';
import BerryScore from '../index';

jest.mock('@/hooks/usePokemon', () => ({
  usePokemon: () => ({
    pokemon: {
      selected: {
        prevExp: 100,
        currentExp: 200,
      },
    },
  }),
}));

jest.mock('@/hooks/useAnimation', () => ({
  useAnimation: () => ({
    animateOpacityToggle: jest.fn(),
  }),
}));

jest.spyOn(global.Math, 'random').mockReturnValue(0.5);

describe('BerryScore component', () => {
  test('matches snapshot', () => {
    const {toJSON} = render(<BerryScore />);
    expect(toJSON()).toMatchSnapshot();
  });

  afterAll(() => {
    jest.spyOn(global.Math, 'random').mockRestore();
  });
});
