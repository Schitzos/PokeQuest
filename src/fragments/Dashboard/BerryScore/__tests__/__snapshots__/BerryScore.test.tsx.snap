// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerryScore component matches snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, 0.8)",
      "borderRadius": 16,
      "left": 150,
      "opacity": 0,
      "padding": 8,
      "position": "absolute",
      "top": -300,
      "zIndex": 4000,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#52D5BA",
          "fontFamily": "PixeloidMono",
          "fontSize": 24,
          "fontWeight": "normal",
          "textAlign": "left",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    +
    100
  </Text>
</View>
`;
